luncheon/headers.lua

```diff
@@ -76,7 +76,7 @@ function Headers:_handle_single_line(line)
     self._inner[self._last_key] = string.format("%s %s", existing, string.sub(line, 2))
     return 1
   end
-  for raw_key, value in string.gmatch(line, '([^%c()<>@,;:\\"/%[%]?={} \t]+): (.+);?') do
+  for raw_key, value in string.gmatch(line, '([^%c()<>@,;:\\"/%[%]?={} \t]+):%s*(.-);?$') do
```

Comment:
I think it might make sense to trim this down a bit on the key parsing. Looking at the RFC again

- This entry says that keys match `token`: https://www.rfc-editor.org/rfc/rfc2616#page-32
- `token` is defined here: https://www.rfc-editor.org/rfc/rfc2616#page-17

> 1*<any CHAR except CTLs or separators>

This would seem to indicate that the pattern above is very much incorrect as it should allow for nearly all of the characters in the negative class

Reply:
Good catch. I'll take a look at that.

Reply:
@FreeMasen I think that the current pattern is actually correct minus the fact that the parens (`()`) aren't escaped.

I moved the pattern in to a variable and documented it with the BNF for review.